ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -335,10 +335,24 @@ private void checkCapacityAndUpdateWriteStatusDelegate(long totalCapacity, long
       }
       //else: maintain current replicaId status if percentFilled between threshold - delta and threshold
     } else {
-      logger.debug("The WriteStatusDelegate is not instantiated");
+      logger.debug("The ReplicaStatusDelegate is not instantiated");
     }
   }
 
+  /**
+   * Update the stopped replicas list based on the state of this store.
+   * @param isStopped whether the store is stopped ({@code true}) or started.
+   * @return {@code true} if StoppedReplicas list has been updated successfully.
+   */
+  public boolean updateStoppedReplicasList(boolean isStopped) {
+    if (replicaStatusDelegate != null) {
+      logger.info("Setting replica stopped state via ReplicaStatusDelegate on replica {}", replicaId);
+      return replicaStatusDelegate.setReplicaStoppedState(replicaId, isStopped);
+    }
+    logger.error("The ReplicaStatusDelegate is not instantiated");
```

Comment:
i prefer a single return path - i find it easier to read and understand
```
    boolean updated = false;
    if (replicaStatusDelegate != null) {
      logger.info("Setting replica stopped state via ReplicaStatusDelegate on replica {}", replicaId);
      updated = replicaStatusDelegate.setReplicaStoppedState(replicaId, isStopped);
    } else {
      logger.error("The ReplicaStatusDelegate is not instantiated");
    }
    return updated;
```

Reply:
Done.